### ISSUE TITLE
Changed behavior after update to 0.10.0

### ISSUE THREAD

**Issue by obrie** [pluginaweek/state_machine#80]
**Tomek Wałkuski** opened this issue

Hi,

I have Customer model which has two state machines: :status and :archived (boolean).

For :status machine initial status is set to :enabled, but on 0.9.X I could create customer (through form, REST API and so on) without providing status param: customer had initial status set, but now it is different. I have to explicitly set params[:status] or I get ActiveRecord::RecordInvalid: Validation failed: Status is invalid exception.

[original LH ticket](http://pluginaweek.lighthouseapp.com/projects/13288/tickets/70)

 This ticket has 0 attachment(s).

**Comment by obrie** [comment]
**Jens-Christian Fischer** commented

same here for more a model with multiple state machines. I got around this by creating a before_validation callback:

```
before_validation :set_sub_states

def set_sub_states
  if self.new_record?
    self.sub_state = "new"
  end
end
```

**Comment by obrie** [comment]
(from [a6cb505298b2a6c8914b22331cc908eb2f8692b6]) Fix classes with multiple state machines failing to initialize in ActiveRecord / Mongoid / Sequel integrations [#70 state:resolved]
https://github.com/pluginaweek/state_machine/commit/a6cb505298b2a6c8914b22331cc908eb2f8692b6

**Comment by obrie** [comment]
Hey guys -

Thanks for the heads up!  This was a big miss on my part... I apologize for that.  I&rsquo;ve released 0.10.1 to fix this issue.

**Comment by obrie** [comment]
**qoobaa** commented

Thanks for solving the issue! It works perfectly now.
